### PR TITLE
61 create eventlist and eventlistitem components

### PR DESCRIPTION
**Pull request by ethanhmaness**
# Description
Added EventList and EventListItem components to display information about events in a list. Icons are stored in a separate file "icons.tsx" so they can be used elsewhere with consistency.

## Relevant issue(s)

#61 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have assigned reviewers to this PR

### CODE REVIEW

File: src/utils/icons.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import MasksIcon from '@mui/icons-material/Masks';
+import ChildFriendlyIcon from '@mui/icons-material/ChildFriendly';
+import { ReactNode } from 'react';
+
+// SVG icons for each event role
+export const roleIcons: { [id: string]: ReactNode } = {
+  ['Food']: <LocalDiningIcon sx={{ color: '#F5A81C' }} />,
+  ['Medical']: <LocalHospitalIcon sx={{ color: 'red' }} />,
+  ['Dental']: <MasksIcon sx={{ color: 'lightblue' }} />,
+  ['Save the Babies']: <ChildFriendlyIcon sx={{ color: 'lightgray' }} />,
```
Comment (zaviermiller) [inline src/utils/icons.tsx:12]: i dont think you need to surround this in `[]`, so it could be `"Save the Babies": ...`

Comment (ethanhmaness) [reply src/utils/icons.tsx]: fixed

File: src/utils/icons.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import MasksIcon from '@mui/icons-material/Masks';
+import ChildFriendlyIcon from '@mui/icons-material/ChildFriendly';
+import { ReactNode } from 'react';
+
+// SVG icons for each event role
+export const roleIcons: { [id: string]: ReactNode } = {
+  Food: <LocalDiningIcon sx={{ color: '#F5A81C' }} />,
+  Medical: <LocalHospitalIcon sx={{ color: 'red' }} />,
+  Dental: <MasksIcon sx={{ color: 'lightblue' }} />,
+  'Save the Babies': <ChildFriendlyIcon sx={{ color: 'lightgray' }} />,
```
Comment (zaviermiller) [inline src/utils/icons.tsx:12]: this is actually what generates the warnings. these need a `key` prop.

File: src/app/components/EventList/EventListItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { EventResponse } from '@/types/dataModel/event';
```
Comment (zaviermiller) [inline src/app/components/EventList/EventListItem/index.tsx:1]: the path should be `src/components/...` not `src/app/components/...`

File: src/app/components/EventList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { List } from '@mui/material';
```
Comment (zaviermiller) [inline src/app/components/EventList/index.tsx:1]: the path should be `src/components/...` not `src/app/components/...`

File: src/utils/icons.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import LocalHospitalIcon from '@mui/icons-material/LocalHospital';
+import MasksIcon from '@mui/icons-material/Masks';
+import ChildFriendlyIcon from '@mui/icons-material/ChildFriendly';
+import { ReactNode } from 'react';
+
+// SVG icons for each event role
+export const roleIcons: { [id: string]: ReactNode } = {
+  Food: <LocalDiningIcon sx={{ color: '#F5A81C' }} />,
+  Medical: <LocalHospitalIcon sx={{ color: 'red' }} />,
+  Dental: <MasksIcon sx={{ color: 'lightblue' }} />,
```
Comment (zaviermiller) [inline src/utils/icons.tsx:11]: take a look at [this](https://github.com/hack4impact-utk/compassion-ministries/pull/121/files#diff-00a3a399ceb9cee803e9a3a05f2d7c296e7a9ade7798aedcf23b2ba769fcfd6dR25-R36). This is how @Frank-D-stein implemented the icons, and will ensure that no matter the order of the roles in the database that they will appear in a consistent order.

**Review by zaviermiller — CHANGES_REQUESTED** [review]
look at comments, looks very good tho

**Review by zaviermiller — APPROVED** [review]

**Pull request merged**